dali/test/python/test_dltensor_operator.py

```diff
@@ -153,13 +183,154 @@ def mxnet_cast(in1, in2):
     return [mxnd.cast(t, dtype='float32') for t in in1], [mxnd.cast(t, dtype='int64') for t in in2]
 
 
-def test_mxnet_flatten():
-    mxnet_case(mxnet_flatten)
+def test_mxnet_flatten_cpu():
+    mxnet_case(mxnet_flatten, device='cpu')
+
+
+def test_mxnet_flatten_gpu():
+    mxnet_case(mxnet_flatten, device='gpu')
 
 
-def test_mxnet_slice():
-    mxnet_case(mxnet_slice)
+def test_mxnet_slice_cpu():
+    mxnet_case(mxnet_slice, device='cpu')
+
+
+def test_mxnet_slice_gpu():
+    mxnet_case(mxnet_slice, device='gpu')
 
 
 def test_mxnet_cast():
-    mxnet_case(mxnet_cast)
+    mxnet_case(mxnet_cast, device='cpu')
+
+
+cupy_stream = cupy.cuda.Stream()
+
+
+def cupy_adapter_sync(fun, in1, in2):
+    with cupy_stream:
+        tin1 = [cupy.fromDlpack(dltensor) for dltensor in in1]
+        tin2 = [cupy.fromDlpack(dltensor) for dltensor in in2]
+        tout1, tout2 = fun(tin1, tin2)
+        out1, out2 = [tout.toDlpack() for tout in tout1], \
+                     [tout.toDlpack() for tout in tout2]
+    cupy_stream.synchronize()
+    return out1, out2
+
+
+def cupy_adapter(fun, in1, in2):
+    tin1 = [cupy.fromDlpack(dltensor) for dltensor in in1]
+    tin2 = [cupy.fromDlpack(dltensor) for dltensor in in2]
+    tout1, tout2 = fun(tin1, tin2)
+    return [tout.toDlpack() for tout in tout1], \
+           [tout.toDlpack() for tout in tout2]
+
+
+def cupy_wrapper(fun, synchronize):
+    if synchronize:
+        return lambda in1, in2: cupy_adapter_sync(fun, in1, in2)
+    else:
+        return lambda in1, in2: cupy_adapter(fun, in1, in2)
+
+
+def cupy_case(fun, synchronize=True):
+    load_pipe = LoadingPipeline('gpu')
+    op_pipe = DLTensorOpPipeline(cupy_wrapper(fun, synchronize), 'gpu', synchronize)
+
```

Comment:
you can use `compare_pipelines`:
```
    from test_utils import compare_pipelines
    compare_pipelines(LoadingPipeline('gpu'),
                                   DLTensorOpPipeline(cupy_wrapper(fun, synchronize), 'gpu', synchronize),
                                   batch_size=BATCH_SIZE, N_iterations=ITERS, eps=1e-07)
```

Reply:
It's not going to work. I compare the result of the DlTensorOpPipeline with manually post-processed data. `compare_pipelines` doesn't have such functionality.